Assets/MixedRealityToolkit/Inspectors/Profiles/DataProviderAccessServiceInspector.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Abstract class providing base functionality for data provider management in inspector. Useful for core systems that follow dataprovider access model.
+    /// </summary>
+    public abstract class BaseDataProviderServiceInspector : BaseMixedRealityToolkitConfigurationProfileInspector
+    {
+        protected class ServiceConfigurationProperties
+        {
+            public SerializedProperty componentName;
```

Comment:
These should be [private fields and/or public properties](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#encapsulation)

Reply:
Even for protected/private classes?

Reply:
I'd vote yes, since I prefer we treat all code in the repo to the same standard, but am happy to open it up for discussion!

Protected classes / methods show up in our docs too (though this is a bad example, since it has public fields...might need to do an update pass on the repo): https://microsoft.github.io/MixedRealityToolkit-Unity/api/Microsoft.MixedReality.Toolkit.Editor.MixedRealityWireframeShaderGUI.Styles.html

Reply:
Fair enough. I'm going to cheat here though. I've switched them to internal since this is an inspector class and meant as an internal data container

Reply:
Works for me!